fdb-record-layer-core/src/main/java/com/apple/foundationdb/record/provider/foundationdb/SubspaceProvider.java

```diff
@@ -23,28 +23,44 @@
 import com.apple.foundationdb.annotation.API;
 import com.apple.foundationdb.record.logging.LogMessageKeys;
 import com.apple.foundationdb.subspace.Subspace;
-
 import javax.annotation.Nonnull;
 import java.util.concurrent.CompletableFuture;
 
 /**
  * Subspace provider can provide a subspace (might be blocking) and logging information to the subspace (non-blocking).
  */
-@API(API.Status.MAINTAINED)
+@API(API.Status.INTERNAL)
 public interface SubspaceProvider {
     /**
      * This might be blocking if the subspace is never fetched before.
+     *
+     * @param context record context used to resolve the subspace.
+     *
      * @return Subspace
      */
     @Nonnull
-    Subspace getSubspace();
+    Subspace getSubspace(@Nonnull FDBRecordContext context);
 
+    /**
+     * @param context record context used to resolve the subspace.
+     *
+     * @return CompletableFuture&lt;Subspace&gt;
+     */
     @Nonnull
-    CompletableFuture<Subspace> getSubspaceAsync();
+    CompletableFuture<Subspace> getSubspaceAsync(@Nonnull FDBRecordContext context);
 
     @Nonnull
     LogMessageKeys logKey();
 
+    /**
+     * @param context record context used to resolve the subspace. Implementations might use
+     * this to resolve a logical database path representation of a subspace to its corresponding
+     * stored representation in order to facilitate logging operations.
```

Comment:
I think most of this would make more sense in the body of the Javadoc, and then the parameter description can just be the shorter phrase describing what the context is.